src/lib/SX1280Driver/SX1280_hal.cpp

```diff
@@ -0,0 +1,366 @@
+    pinMode(GPIO_PIN_RX_ENABLE, OUTPUT);
+    digitalWrite(GPIO_PIN_RX_ENABLE, LOW);
+#endif
+
+#if defined(GPIO_PIN_TX_ENABLE)
+    pinMode(GPIO_PIN_TX_ENABLE, OUTPUT);
+    digitalWrite(GPIO_PIN_TX_ENABLE, LOW);
+#endif
+
+#ifdef PLATFORM_ESP32
+    SPI.begin(GPIO_PIN_SCK, GPIO_PIN_MISO, GPIO_PIN_MOSI, -1); // sck, miso, mosi, ss (ss can be any GPIO)
+    SPI.setFrequency(18000000);
+#endif
+
+#ifdef PLATFORM_ESP8266
+    Serial.println("PLATFORM_ESP8266");
+    SPI.begin();
+    //SPI.pins(this->SX1280_SCK, this->SX1280_MISO, this->SX1280_MOSI, -1);
+    SPI.setBitOrder(MSBFIRST);
+    SPI.setDataMode(SPI_MODE0);
+    SPI.setFrequency(18000000);
+#endif
+
+#ifdef PLATFORM_STM32
+    SPI.setMOSI(GPIO_PIN_MOSI);
+    SPI.setMISO(GPIO_PIN_MISO);
+    SPI.setSCLK(GPIO_PIN_SCK);
+    SPI.setBitOrder(MSBFIRST);
+    SPI.setDataMode(SPI_MODE0);
+    SPI.begin();
+    SPI.setClockDivider(SPI_CLOCK_DIV4); // 72 / 8 = 9 MHz
+#endif
+
+    //attachInterrupt(digitalPinToInterrupt(GPIO_PIN_BUSY), this->busyISR, CHANGE); //not used atm
+    attachInterrupt(digitalPinToInterrupt(GPIO_PIN_DIO1), this->dioISR, RISING);
+}
+
+void ICACHE_RAM_ATTR SX1280Hal::reset(void)
+{
+    Serial.println("SX1280 Reset");
+    delay(50);
+    digitalWrite(GPIO_PIN_RST, LOW);
+    delay(50);
+    digitalWrite(GPIO_PIN_RST, HIGH);
+
+    while (digitalRead(GPIO_PIN_BUSY) == HIGH) // wait for busy
+    {
+#ifdef PLATFORM_STM32
+        __NOP();
+#elif PLATFORM_ESP32
+        _NOP();
+#elif PLATFORM_ESP8266
+        _NOP();
+#endif
+    }
+
+    //this->BusyState = SX1280_NOT_BUSY;
+    Serial.println("SX1280 Ready!");
+}
+
+void ICACHE_RAM_ATTR SX1280Hal::WriteCommand(SX1280_RadioCommands_t command, uint8_t val)
+{
+    WaitOnBusy();
+    digitalWrite(GPIO_PIN_NSS, LOW);
+
+    SPI.transfer((uint8_t)command);
+    SPI.transfer(val);
+
+    digitalWrite(GPIO_PIN_NSS, HIGH);
+}
+
+void ICACHE_RAM_ATTR SX1280Hal::WriteCommand(SX1280_RadioCommands_t command, uint8_t *buffer, uint8_t size)
+{
+    WORD_ALIGNED_ATTR uint8_t OutBuffer[size + 1];
+
+    OutBuffer[0] = (uint8_t)command;
+    memcpy(OutBuffer + 1, buffer, size);
+
+    WaitOnBusy();
+    digitalWrite(GPIO_PIN_NSS, LOW);
+    SPI.transfer(OutBuffer, (uint8_t)sizeof(OutBuffer));
+    digitalWrite(GPIO_PIN_NSS, HIGH);
+}
+
+void ICACHE_RAM_ATTR SX1280Hal::ReadCommand(SX1280_RadioCommands_t command, uint8_t *buffer, uint8_t size)
+{
+    WORD_ALIGNED_ATTR uint8_t OutBuffer[size + 2];
+
+    WaitOnBusy();
+    digitalWrite(GPIO_PIN_NSS, LOW);
+
+    if (command == SX1280_RADIO_GET_STATUS)
+    {
+        OutBuffer[0] = (uint8_t)command;
+        OutBuffer[1] = 0x00;
+        OutBuffer[2] = 0x00;
+        SPI.transfer(OutBuffer, 3);
+        buffer[0] = OutBuffer[0];
+    }
+    else
+    {
+        OutBuffer[0] = (uint8_t)command;
+        OutBuffer[1] = 0x00;
+        memcpy(OutBuffer + 2, buffer, size);
+        SPI.transfer(OutBuffer, sizeof(OutBuffer));
+        memcpy(buffer, OutBuffer + 2, size);
```

Comment:
And this?

Reply:
no same deal, write with an offset and only copy in the data part which is 2 less bytes